### ISSUE TITLE
Buildozer: changing attributes in override functions of MODULE.bazel

### ISSUE THREAD

**Issue by linzhp** [bazelbuild/buildtools#1218]
Buildozer is not able to recognize the functions with `module_name` in MODULE.bazel file. So `buildozer print //MODULE.bazel:all` would print something like:

```
main module
rules_python bazel_dep
rules_python_gazelle_plugin bazel_dep
gazelle bazel_dep
hermetic_cc_toolchain bazel_dep
bazel_skylib bazel_dep
rules_proto_grpc bazel_dep
 archive_override
rules_proto_grpc_python bazel_dep
 archive_override
```
Note that the `archive_override` doesn't have name. So we can't update them from command line.

We can follow the convention of Bazel to create a target `rules_proto_grpc~override` when Buildozer sees a *_override rule with `module_name = "rules_proto_grpc"`.

**Comment by tyler-french** [comment]
We may want to consider having a different `attr` marked as the "identifier" for different rule kinds.

For example, with gazelle overrides, we use `path` as the identifier.

That way these could be identifiable:

```
go_deps.from_file
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.gazelle_override
 go_deps.module_override
 go_deps.module_override
 go_deps.module_override
 go_deps.module_override
 go_deps.module_override
 go_deps.module_override
 go_deps.module
 go_deps.archive_override
 go_deps.module
 go_deps.archive_override
 go_deps.module
 go_deps.archive_override
```

**Comment by fmeum** [comment]
I think we want new buildozer syntax here, maybe `//some/pkg:some_attr=foo`?

**Comment by linzhp** [comment]
Do we need new syntax? I was thinking `buildozer "set some_attr foo" //MODULE.bazel:rules_proto_grpc~override`. Does that work?

**Comment by fmeum** [comment]
That feels quite magic and ties Buildozer to an implementation detail of Bazel (canonical repository names). It also doesn't allow specifying extension tags such as `go_deps.gazelle_override`.

**Comment by linzhp** [comment]
are you suggesting something like `buildozer "set some_attr foo" //MODULE.bazel:path=golang.org/x/tools`

**Comment by fmeum** [comment]
> are you suggesting something like `buildozer "set some_attr foo" //MODULE.bazel:path=golang.org/x/tools`

Yes, exactly that, where `:name=foo` would match the current behavior. There could be other syntax for this, but the functionality of matching by a custom attribute is the extension I would like best.

**Comment by albertocavalcante** [comment]
Is this the latest on this? I was looking into `buildozer` syntax to manipulate  a `MODULE.bazel` file (e.g. include or change a `go_deps.module`)
